cobbler/settings/migrations/V3_0_1.py

```diff
@@ -38,14 +38,47 @@ def normalize(settings: dict) -> dict:
     return schema.validate(settings)
 
 
+def __migrate_modules_conf():
+    modules_conf_path = "/etc/cobbler/modules.conf"
+    with open(modules_conf_path, "r") as modules_conf_file:
+        modules_conf = modules_conf_file.readlines()
+
+    modules_conf_result = []
+    for line in modules_conf:
+        authn_index = line.find("authn_")
+        if authn_index > -1:
+            modules_conf_result.append(
+                "%sauthentication.%s" % (line[:authn_index], line[(authn_index + 6) :])
+            )
+            continue
+        authz_index = line.find("authz_")
+        if authz_index > -1:
+            modules_conf_result.append(
+                "%sauthorization.%s" % (line[:authz_index], line[(authz_index + 6) :])
+            )
+            continue
+        manage_index = line.find("manage_")
+        if manage_index > -1:
+            modules_conf_result.append(
+                "%smanagers.%s" % (line[:manage_index], line[(manage_index + 7) :])
+            )
+            continue
+        modules_conf_result.append(line)
+
+    with open(modules_conf_path, "w") as modules_conf_file:
+        for line in modules_conf_result:
+            modules_conf_file.write(line)
```

Comment:
I find your version a bit repetitive and propose this alternative form. It does not first read the whole file into memory since that's not needed. My version also does not rely on magic numbers (6 and 7 in your case), which improves readability. I executed it locally and got the same output as your implementation.
```suggestion
    with open("modules.conf", "r") as modules_conf_file:
        result = []
        replacements = {
            "authn_": "authentication.",
            "authz_": "authorization.",
            "manage_": "managers.",
        }
        for line in modules_conf_file:
            for to_replace, replacement in replacements.items():
                idx = line.find(to_replace)
                if idx == -1:
                    continue

                result.append(
                    "%(head)s%(replacement)s%(tail)s"
                    % {
                        "head": line[:idx],
                        "replacement": replacement,
                        "tail": line[idx + len(to_replace) :],
                    }
                )
                break
            else:  # no break occured -> nothing to replace
                result.append(line)
    with open("modules.conf", "w") as modules_conf_file:
        for line in result:
            modules_conf_file.write(line)
```